packages/frontend-v2/components/ScheduleCourse/ScheduleCourse.tsx

```diff
@@ -46,7 +48,7 @@ export const DraggableScheduleCourse: React.FC<
     });
 
   useEffect(() => {
-    if (setIsRemove) setIsRemove(over === null);
+    if (setIsRemove) setIsRemove(over?.id === DELETE_COURSE_AREA_DND_ID);
```

Comment:
I feel like the overlap between the overlay course being dragged and the original course is really confusing for anyone trying to understand the logic behind draggable course. Might be worth seperating them into different components with different props, but have the same styling etc.

Reply:
ALSO, we can actually replace this state with `useMemo` instead. Its like a glorified `useEffect` designed to udpate a value that relies on another piece of state. I think we can start implementing this more and more in our codebase.

Reply:
I agree on the overlap, let me spend some time refactoring!